packages/invoice-dashboard/src/lib/view-requests.svelte

```diff
@@ -76,7 +80,45 @@
   let sortColumn = "timestamp";
 
   $: {
-    signer = wallet?.accounts[0]?.address;
+    if (wagmiConfig) {
+      account = getAccount(wagmiConfig);
+    }
+  }
+
+  $: {
+    if (account?.address) {
+      tick().then(() => {
+        getRequests();
+      });
+    }
+  }
+
+  let unwatchAccount: WatchAccountReturnType | undefined;
+
+  onMount(() => {
+    unwatchAccount = watchAccount(wagmiConfig, {
+      onChange(data) {
+        if (data?.address !== account?.address) {
+          account = data;
+          if (account?.address) {
+            getRequests();
+          } else {
+            requests = [];
+            activeRequest = undefined;
+            previousWalletAddress = undefined;
+            previousNetwork = undefined;
+          }
+        }
+      },
+    });
+  });
+
+  onDestroy(() => {
+    if (unwatchAccount) unwatchAccount();
```

Comment:
This conditional makes `unwatchAccount` seem like a boolean. Recommend adjusting to make it clear that we're checking for existence of a function.

```suggestion
    if (typeof unwatchAccount === 'function') unwatchAccount();
```